ts/session/apis/open_group_api/sogsv3/knownBlindedkeys.ts

```diff
@@ -0,0 +1,413 @@
+import { from_hex, to_hex } from 'libsodium-wrappers-sumo';
+import _, { cloneDeep, flatten, isEmpty, isEqual, isString, uniqBy } from 'lodash';
+import { getConversationController } from '../../../conversations';
+import { LibSodiumWrappers } from '../../../crypto';
+import { KeyPrefixType, PubKey } from '../../../types';
+import { crypto_sign_curve25519_pk_to_ed25519 } from 'curve25519-js';
+import { Data } from '../../../../data/data';
+import { combineKeys, generateBlindingFactor } from '../../../utils/SodiumUtils';
+import { OpenGroupData } from '../../../../data/opengroups';
+import { ConversationModel } from '../../../../models/conversation';
+import { UserUtils } from '../../../utils';
+import { SogsBlinding } from './sogsBlinding';
+import { fromHexToArray } from '../../../utils/String';
+import { roomHasBlindEnabled } from './sogsV3Capabilities';
+
+export type BlindedIdMapping = {
+  blindedId: string;
+  serverPublicKey: string;
+  realSessionId: string;
+};
+
+export const KNOWN_BLINDED_KEYS_ITEM = 'KNOWN_BLINDED_KEYS_ITEM';
+
+// for now, we assume we won't find a lot of blinded keys.
+// So we can store all of those in a single JSON string in the db.
+let cachedKnownMapping: Array<BlindedIdMapping> | null = null;
```

Comment:
Maybe we should future proof this?

Reply:
I am unsure if there is really a need. I would say we easily break it apart with a migration if needed and if some users have performance bottlenecks because of it, but I would say we will be fine. 
We only load it once on the app start, and will only write the data when we find a match that we haven't yet in the blinding logic, which does not happen often